apis/Google.Cloud.Spanner.V1/Google.Cloud.Spanner.V1/TransactionPool.cs

```diff
@@ -243,40 +245,47 @@ public void MarkTransactionUsed()
                 }
             }
 
-            public async Task CreateTransactionAsync(Session session, TransactionOptions options)
+            public Task CreateTransactionAsync(Session session, TransactionOptions options)
```

Comment:
you typically would not call this method directly simultaneously from different threads.
You can call methods on SpannerConnection concurrently.  If you call begintransaction there, it will allocate a separate session for each call and then call createtransaction on each separately.